flathunter/abstract_crawler.py

```diff
@@ -68,10 +44,26 @@ def get_page(self, search_url, page_no=None):
 
     def get_soup_from_url(self, url):
         """Creates a Soup object from the HTML at the provided URL"""
-        self.rotate_user_agent()
-        resp = requests.get(url, headers=self.HEADERS)
-        if resp.status_code != 200:
-            self.__log__.error("Got response (%i): %s", resp.status_code, resp.content)
+        proxies_list = proxies.get_proxies()
```

Comment:
I like the option to use a list of proxies - if it helps with idealista, that's great. I notice that for the portals in Germany that are blocking with reCaptcha, running with the proxy doesn't help. It would be great if this could be optional. You can add a section to the config file (e.g. use_proxy_list: true), and add a function to the Config object `def use_proxy(self): return ("use_proxy_list" in self.config and self.config["use_proxy_list"])`.

Reply:
Of couse, making it optional would be in fact much cleaner!